### DISCUSSION TITLE
Massive overestimation of absolute permeability in extracted networks from Nano-CT Tiff stacks

### DISCUSSION THREAD

**Discussion by andtoni** [PMEAL/OpenPNM#2926] [Q&A] [answered]
_(Note, you don't have to fill out every section here. They're just here for guidance. That said, nicely detailed feature requests are more likely to get dev attention sooner)_

**Describe the bug**
Hi all, I would sincerely appreciate some advice.

I start off with importing a 800^3 tiff stack, converting to 3d tiff, segmenting and finally binarising all within my python code.

I then run the snow algorithm, and convert the extracted network using io.network_from_porespy.

Interestingly, the new network lacks surface labels, so I used the topotools.label_faces to assign surface labels, however, if I leave the tol at 0, it only labels a single pore per surface. If I increase the tol to 0.1, this number increases dramatically. Not sure how to approach this step correctly.

I then am able to trim clusters and run the absolute permeability as instructed in the docs, however, my results cause many errors and output extremely high mD values which do not seem possible. I would sincerely appreciate any advice around this.

![Screenshot 2024-07-08 150132](https://github.com/PMEAL/OpenPNM/assets/174879741/39398be6-fa64-41ce-a316-9d36e14f9c31)

**Comment by jgostick** [comment]
Hi @andtoni, I am going to remove the bug label and transfer this over to the Discussion's section because I'm fairly sure this falls under the category of 'needing help' rather than 'bug'.

**Comment by jgostick** [comment, marked as answer]
Hi @andtoni
My first guess for the over estimation is that you may not have entered the voxel size into the `snow2` algorithm?  If you don't do this then it defaults to `1.0` which is perhaps a bad default because it basically means the voxels are 1 meter.  

The issue regarding the errors you're getting is because you are apply the find_domain_area/length functions to an extracted networks. In extracted networks the pores are randomly located so they are obviously not going to be co-planar.  Assuming you used `snow2` the network extraction function will labelled pores as 'xmin', 'xmax' etc that you can use as boundaries.  You say your network is lacking surface labels, but I am pretty sure that `snow2` always assigns surface labels, so they should be there.  Could you paste the result of `print(pn)` here?

**Reply by ma-sadeghi** [reply]
Maybe we should make voxel size a mandatory kwarg?

**Reply by jgostick** [reply]
Yep, will add this to bug tracker on PoreSpy

**Comment by andtoni** [comment]
Dear Prof @jgostick 

Thank you for taking your time to respond to this, I apologise for mislabelling the issue.

Your guess was 100% correct, my voxel size was set to 1 meter rather then 0.54 microns, I seem to be getting much more reasonable responses now. My networks do have the 'xmin' labels etc via the snow2, I will be sure to use those.
